lib/rollbar/plugins/basic_socket.rb

```diff
@@ -1,16 +1,29 @@
 Rollbar.plugins.define('basic_socket') do
+  load_on_demand
+
   dependency { !configuration.disable_core_monkey_patch }
 
   # Needed to avoid active_support (< 4.1.0) bug serializing JSONs
-  dependency { defined?(ActiveSupport::VERSION::STRING) }
+  dependency do
+    defined?(ActiveSupport::VERSION::STRING) &&
+      Gem::Version.new(ActiveSupport::VERSION::STRING) < Gem::Version.new('5.2.0')
+  end
+
+  @original_as_json = ::BasicSocket.public_instance_method(:as_json)
```

Comment:
With the other error out of the way, this now fails for ActiveSupport <= 4.0 which doesn't define an `#as_json` method. 

The original code expected `dependency { defined?(ActiveSupport::VERSION::STRING) }` to exclude ActiveSupport < 4.1, but that doesn't seem to be working.

Reply:
Fixed